### ISSUE TITLE
Avoid pulling data out of tx_index.db in EthGetBlockByNumber / EthGetBlockByHash

### ISSUE THREAD

**Issue by enlight** [loomnetwork/loomchain#1443]
`EthGetBlockByNumber` and `EthGetBlockByHash` call `GetBlockByNumber` which attempts to load data from `tx_index.db`
https://github.com/loomnetwork/loomchain/blob/141380a5e0c157f3492774f74e960c51dcfaa0e5/eth/query/block.go#L84

On some production clusters that `tx_index.db` may be missing data so `EthGetBlockByNumber` & `EthGetBlockByNumber` fail with an error. We don't actually need to rely on `tx_index.db` in this case because the tx result data we need can be obtained from `TendermintBlockStore.GetBlockResults` (which pulls data from `state.db`). Furthermore the data in question is only needed for EVM txs, so we should avoid retrieving it.

In short in `GetBlockByNumber` for an EVM tx try to get data from `blockStore.GetBlockResults` first, then fall back to `blockStore.GetTxResult` if necessary.

**Comment by enlight** [comment]
This issue can be reproduced using:
```
curl -X POST --data '{"jsonrpc":"2.0","method":"eth_getBlockByNumber","id":1,"params":["0x5", true]}' https://plasma.dappchains.com/eth
```

**Comment by enlight** [comment]
Resolved by https://github.com/loomnetwork/loomchain/pull/1446
